### PR TITLE
Mac M1 build update.

### PR DESCRIPTION
**Pull request by devnexen**
This suppports the crc32 word size various flavors instructions.

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -7520,7 +7520,7 @@ FORCE_INLINE void _mm_free(void *addr)
 // https://msdn.microsoft.com/en-us/library/bb514036(v=vs.100)
 FORCE_INLINE uint32_t _mm_crc32_u8(uint32_t crc, uint8_t v)
 {
-#if defined(__aarch64__) && defined(__ARM_FEATURE_CRC32)
+#if defined(__aarch64__) && (defined(__ARM_FEATURE_CRC32) || defined(__APPLE__))
```
Comment (jserv) [inline sse2neon.h:7523]: I have no idea why you would make this change. The command I executed on Apple M1 to retrieve predefined macros:
```shell
clang -x c /dev/null -dM -E | grep __ARM_FEATURE_CRC32
```

I got the message `#define __ARM_FEATURE_CRC32 1`
It seems that the above change is not necessary. Can you clarify?

Comment (devnexen) [reply sse2neon.h:7523]: Sure but in the project it was used due to compile flags, it was not defined.

Comment (jserv) [reply sse2neon.h:7523]: Can you show me some examples which rely on `__APPLE__` macro instead of `__ARM_FEATURE_CRC32`?

**Comment by devnexen** [comment]
I think I ll just attempt to fix the project build instead.

**Pull request closed without merging**